jerry-core/ecma/base/ecma-helpers-value.c

```diff
@@ -136,9 +160,8 @@ ecma_is_value_null (ecma_value_t value) /**< ecma value */
 inline bool __attr_pure___ __attr_always_inline___
 ecma_is_value_boolean (ecma_value_t value) /**< ecma value */
 {
-  return (ecma_get_value_type_field (value) == ECMA_TYPE_SIMPLE
-          && (ecma_get_value_value_field (value) == ECMA_SIMPLE_VALUE_TRUE
-              || ecma_get_value_value_field (value) == ECMA_SIMPLE_VALUE_FALSE));
+  return (ecma_is_value_equal_to_simple_value (value, ECMA_SIMPLE_VALUE_TRUE)
+          || ecma_is_value_equal_to_simple_value (value, ECMA_SIMPLE_VALUE_FALSE));
```

Comment:
If you'd stay with the now-proposed logic, this could simply be `ecma_is_value_true (value) || ecma_is_value_false (value)`. The original version made use of the fact the checking the "simple" property was done only once. If you'd like to do some similar optimization with the current approach, then it would be somethin like:
```c
const ecma_value_t v = value | ECMA_VALUE_ERROR_FLAG;
return  (v == (ECMA_SIMPLE_VALUE_TO_VALUE (ECMA_SIMPLE_VALUE_TRUE) | ECMA_VALUE_ERROR_FLAG))
  || (v == (ECMA_SIMPLE_VALUE_TO_VALUE (ECMA_SIMPLE_VALUE_FALSE) | ECMA_VALUE_ERROR_FLAG));
```
As this looks overly complicated and we expect the compiler to inline and optimize all these tiny functions away, I'd suggest the first approach.